src/libraries/System.Private.CoreLib/src/System/Math.cs

```diff
@@ -163,24 +165,50 @@ public static double BitIncrement(double x)
             return BitConverter.Int64BitsToDouble(bits);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static unsafe double CopySign(double x, double y)
         {
-            // This method is required to work for all inputs,
-            // including NaN, so we operate on the raw bits.
+            const long signMask = 1L << 63;
 
-            long xbits = BitConverter.DoubleToInt64Bits(x);
-            long ybits = BitConverter.DoubleToInt64Bits(y);
+            if (Sse.X64.IsSupported)
```

Comment:
Perhaps, can be also optimized for arm64? via `bit`

Reply:
Yep, this code was originally written iirc before arm intrinsics were stable enough to use, and before i had any experience with them. I can prototype a version now, althought I don't have an arm system to benchmark on

Reply:
`BIT` appears to not yet be exposed, I'll ask tanner if it is already covered by an issue, and if not, I'll make one

Reply:
I think it's `AdvSimd.BitwiseSelect`

Reply:
I believe that is `BSL` not `BIT`, but it might be usable for the same function

Reply:
We don't expose BIT or BIF because they are just codegen optimizations of BSL

Reply:
@john-h-k see https://github.com/dotnet/runtime/blob/2ff59a12aa8f91bc24a93e5e6ec8da941e3e5bef/src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp#L315-L330

Reply:
Added 👍